File: frontend/src/api/teacher.ts
import api from './index'

export interface Teacher {
  id: number
  name: string
  phone: string
  campuses: any[]
  status: string
  createdAt: string
}

export const teacherApi = {
  getAll: () => api.get('/teachers'),
  getActive: () => api.get('/teachers/active'),
  getById: (id: number) => api.get(`/teachers/${id}`),
  getByCampus: (campusId: number) => api.get(`/teachers/campus/${campusId}`),
  create: (data: Partial<Teacher>) => api.post('/teachers', data),
  update: (id: number, data: Partial<Teacher>) => api.put(`/teachers/${id}`, data),
  delete: (id: number) => api.delete(`/teachers/${id}`)
}

File: backend/src/main/java/com/edu/management/entity/Teacher.java
package com.edu.management.entity;

import com.edu.management.enums.TeacherStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

@Getter
@Setter
@ToString(exclude = "campuses")
@Entity
@Table(name = "teachers")
public class Teacher {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, length = 50)
    private String name;
    
    @Column(nullable = false, unique = true, length = 20)
    private String phone;
    
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
        name = "teacher_campus",
        joinColumns = @JoinColumn(name = "teacher_id"),
        inverseJoinColumns = @JoinColumn(name = "campus_id")
    )
    private Set<Campus> campuses = new HashSet<>();
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private TeacherStatus status = TeacherStatus.ACTIVE;
    
    @CreationTimestamp
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    private LocalDateTime updatedAt;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Teacher teacher = (Teacher) o;
        return Objects.equals(id, teacher.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}
